java/org/contikios/cooja/Simulation.java

```diff
@@ -53,13 +51,26 @@
  *
  * @author Fredrik Osterlind
  */
-public class Simulation extends Observable implements Runnable {
+public class Simulation extends Observable {
+
+  /** Commands sent to the simulation thread to start, stop, or shutdown the simulation */
+  private enum Command {
+    START, STOP, QUIT
+  }
+
   public static final long MICROSECOND = 1L;
   public static final long MILLISECOND = 1000*MICROSECOND;
 
+  /* Lock to wait for simulation state changes */
```

Comment:
Make it a javadoc so it's possible to see in IntelliJ when mousing over things, private fields are not shown in the javadoc generated by `./gradlew javadoc` for me.

Reply:
Ok, I updated this.